depthai_examples/src/stereo_inertial_publisher.cpp

```diff
@@ -0,0 +1,277 @@
+// Inludes common necessary includes for development using depthai library
+#include "depthai/depthai.hpp"
+#include <depthai_bridge/BridgePublisher.hpp>
+#include <depthai_bridge/ImuConverter.hpp>
+#include <depthai_bridge/ImageConverter.hpp>
+#include <depthai_bridge/DisparityConverter.hpp>
+
+
+dai::Pipeline createPipeline(bool enableDepth, bool lrcheck, bool extended, bool subpixel, bool rectify, bool depth_aligned, int stereo_fps){
+    dai::Pipeline pipeline;
+
+    auto monoLeft             = pipeline.create<dai::node::MonoCamera>();
+    auto monoRight            = pipeline.create<dai::node::MonoCamera>();
+    auto xoutLeft             = pipeline.create<dai::node::XLinkOut>();
+    auto xoutRight            = pipeline.create<dai::node::XLinkOut>();
+    auto xoutRectifiedLeft    = pipeline.create<dai::node::XLinkOut>();
+    auto xoutRectifiedRight   = pipeline.create<dai::node::XLinkOut>();
+    auto stereo               = pipeline.create<dai::node::StereoDepth>();
+    auto xoutDepth            = pipeline.create<dai::node::XLinkOut>();
+    auto imu                  = pipeline.create<dai::node::IMU>();
+    auto xoutImu              = pipeline.create<dai::node::XLinkOut>();
+
+    // XLinkOut
+    xoutLeft->setStreamName("left");
+    xoutRight->setStreamName("right");
+    xoutRectifiedLeft->setStreamName("rectifiedleft");
+    xoutRectifiedRight->setStreamName("rectifiedright");
+
+    if (enableDepth) {
+        xoutDepth->setStreamName("depth");
+    }
+    else {
+        xoutDepth->setStreamName("disparity");
+    }
+
+    xoutImu->setStreamName("imu");
+
+    //RGB
+    if(enableDepth && depth_aligned){
+        auto camRgb               = pipeline.create<dai::node::ColorCamera>();
+        auto xoutRgb              = pipeline.create<dai::node::XLinkOut>();
+        xoutRgb->setStreamName("rgb");
+        camRgb->setBoardSocket(dai::CameraBoardSocket::RGB);
+        camRgb->setResolution(dai::ColorCameraProperties::SensorResolution::THE_1080_P);
+        // the ColorCamera is downscaled from 1080p to 720p.
+        // Otherwise, the aligned depth is automatically upscaled to 1080p
+        camRgb->setIspScale(2, 3);
+        // For now, RGB needs fixed focus to properly align with depth.
+        // This value was used during calibration
+        camRgb->initialControl.setManualFocus(135);
+        camRgb->isp.link(xoutRgb->input);
+    }
+
+    // MonoCamera
+    monoLeft->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
+    monoLeft->setBoardSocket(dai::CameraBoardSocket::LEFT);
+    monoLeft->setFps(stereo_fps);
+    monoRight->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
+    monoRight->setBoardSocket(dai::CameraBoardSocket::RIGHT);
+    monoRight->setFps(stereo_fps);
+
+    // StereoDepth
+    stereo->initialConfig.setConfidenceThreshold(200);
+    stereo->setRectifyEdgeFillColor(0); // black, to better see the cutout
+    stereo->initialConfig.setLeftRightCheckThreshold(1);
```

Comment:
set this to 10. We saw better results with 10.

Reply:
Which one I should set to 10? ConfidenceThreshold or LeftRightCheckThreshold or both?

Reply:
sorry my bad. `LeftRightCheckThreshold` to 10. And also set `ConfidenceThreshold` 220

Reply:
Fixed.